packages/api/src/v2/types.ts

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+"use strict";
+
+import { Platform, VsCodeEnv } from "../constants";
+import { UserInteraction } from "../qm/ui";
+import { LogProvider, TelemetryReporter } from "../utils";
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+export type Void = {};
+export const Void = {};
+
+/**
+ * environment meta data
+ */
+export interface EnvMeta {
+  name: string;
+  local: boolean;
+  sideloading: boolean;
+}
+
+export type Json = Record<string, unknown>;
+export type PluginName = string;
+
+/**
+ * project static setting
+ */
+export interface ProjectSetting extends Json {
+  name: string;
+  environments: Record<string, EnvMeta>;
+  currentEnv: string;
+  solution: {
+    name: string;
+    version?: string;
+  };
+  solutionSetting: SolutionSetting;
+}
+
+export interface SolutionSetting extends Json {
+  resourcePlugins: string[];
+}
+
+export interface Inputs extends Json {
+  stage: Stage;
```

Comment:
For Core's API, the only input is:
`createProject: (systemInputs: Inputs) => Promise<Result<string, FxError>>;`
Suggest projectPath and platform move to Inputs

Reply:
Makes sense. Done